src/Listeners/FileListener.php

```diff
@@ -82,53 +83,52 @@ public function postPersist(LifecycleEventArgs $args)
 	{
 		$entity = $args->getEntity();
 
-		if (!$entity instanceof $this->fileEntityClass) {
+		if (!$entity instanceof FileEntity) {
 			return;
 		}
 
 		if ($entity->hasTemporaryFile()) {
-			$entity
-				->setPath($this->dataDir)
-				->setUrl($this->dataUrl)
-				->saveFile();
+			$entity->setBaseDirectoryPath($this->dataDir);
+			$entity->setBaseDirectoryUrl($this->dataUrl);
+			$entity->saveFile();
 
-			$this->em->flush($entity);
+			$this->em->flush();
 
 			if ($entity->getOnAfterSave()) {
 				call_user_func($entity->getOnAfterSave(), $entity);
 			}
 		}
 	}
 
-	public function preRemove(\Doctrine\ORM\Event\LifecycleEventArgs $eventArgs)
+	public function preRemove(LifecycleEventArgs $eventArgs)
 	{
 		$entity = $eventArgs->getEntity();
-
-		if (!$entity instanceof $this->fileEntityClass) {
+		if (!$entity instanceof FileEntity) {
 			return;
 		}
 
-		$this->file = $this->em->createQueryBuilder()
-			->select('e')
-			->from($this->fileEntityClass, 'e')
-			->where('e.id = :id')
-			->setParameter('id', $entity->getId())
-			->getQuery()
-			->getResult('readOnly')[0];
+		// zabespeci nacteni entity pokud jde o proxy, v opacnem pripade
+		// by v post flush nebylo mozne ziskat path lebo by doctrine nenasla entitu v db
+		$entity->getPath();
+
+
+		// smazane file entity si ulozime a pak je v post flush pouzijeme pro smazani jejich soboru
+		// post remove nelze pouzit lebo sa ne vzdy zavola (treba pri smazani pres orphanremoval)
+		// https://github.com/doctrine/orm/issues/6256
+		$this->filesToDelete[] = $entity;
 	}
 
-	public function postRemove(\Doctrine\ORM\Event\LifecycleEventArgs $eventArgs)
+	public function postFlush()
```

Comment:
mozna v tomto byl ten problem, ze ja jsem to delal v postRemove a proto se mi ta iniciovana entita znovu hodila do db... ty kdyz to delas v postFlush tak se to mozna diky tomu nedeje...

Reply:
Puvodne sem to mel v postRemove a taky to slo v pohode.